model/task_queue_service_dependency.go

```diff
@@ -133,57 +133,8 @@ func (d *basicCachedDAGDispatcherImpl) addEdge(fromID string, toID string) error
 	fromNode := d.getNodeByItemID(fromID)
 	toNode := d.getNodeByItemID(toID)
 
+	// The "depend_on" <from> task is not in the DAG so we don't need an edge.
```

Comment:
Is this because it has already been dispatched?

Reply:
It's potentially because it's already been dispatched, or because it doesn't exist.
I'm asserting whatever the reason it's not in the queue we don't need an edge. Given the function of the edge is to sort the queue so the queue is dispatched in the correct order, if a task isn't in the queue we don't need to consider it.